pkg/internal/checkup/checkup.go

```diff
@@ -975,11 +972,83 @@ func (c *Checkup) checkVMIHotplugVolume(ctx context.Context, errStr *string) err
 	return nil
 }
 
-type checkVMIStatusFn func(*kvcorev1.VirtualMachineInstance) (done bool, err error)
+func (c *Checkup) checkConcurrentVMIBoot(ctx context.Context, errStr *string) error {
+	numOfVMs := c.checkupConfig.NumOfVMs
+	log.Printf("checkConcurrentVMIBoot numOfVMs:%d", numOfVMs)
 
-func (c *Checkup) waitForVMIStatus(ctx context.Context, checkMsg string, result, errStr *string, checkVMIStatus checkVMIStatusFn) error {
-	vmName := c.vmUnderTest.Name
+	if c.goldenImagePvc == nil && c.goldenImageSnap == nil {
+		log.Print(MessageSkipNoGoldenImage)
+		c.results.ConcurrentVMBoot = MessageSkipNoGoldenImage
+		return nil
+	}
+
+	var wg sync.WaitGroup
```

Comment:
if you wanted to simplify, you could get rid of sync.WaitGroup and just have 2 loops
- loop over creation
- loop over waiting for ready

definitely not a must, just something to consider

Reply:
I personally prefer the goroutine-way. Looping over creation is less "concurrent", as it's serialized and you will see the last VM created long after the first one when you create for example 100 VMs. Tried that in the first shot but didn't really like this behavior. Looping over the wait for ready is ok (although it's cheaper from the goroutine if we already have it), but since it's serial we won't have the nice indication in the log whenever each VM is ready.

Reply:
right the creation http request will block. nvm